### ISSUE TITLE
Allow ad hoc releases of experimental core plugins 

### ISSUE THREAD

**Issue by Jermolene** [TiddlyWiki/TiddlyWiki5#8477]
It is proposed that plugins whose stability level is "experimental" be allowed to follow their own release schedule, with releases going to production as soon as they are ready.

There are a few changes that will need to be coordinated to make this happen:

* Allow plugins to specify a minimum core version beyond which they will refuse to be installed
* Give plugins a "Release Notes" or "Changelog" tab with a structured display of changes to the plugin
* Devise and document the release procedure (which is essentially copying files under 'plugins/tiddlywiki/*' from 'master' to 'tiddlywiki-com')

For the release notes, the tiddler `$:/plugins/tiddlywiki/pluginname/changelog` would be a data dictionary mapping version numbers to a brief description of the release of the form `2024-08-24 Description of this update`. If plugins wish to add more documentation about a release they can use tiddlers incorporating the version number of the form `$:/plugins/tiddlywiki/pluginname/changelog/v1.0.0` that would contain the full description of the release.

**Comment by pmario** [comment]
I think, that the full description of the plugin-history should be part of a demo-edition.

The plugin itself should only contain the minimal changelog-info. Over time there may be plugins, where the history is much longer than the plugin code itself.

I personally consider this a waste of space. 

eg: My [custom-markup plugin](https://wikilabs.github.io/editions/custom-markup/) has a long list of history. The plugin itself only contains the latest 2 or 3 points and a link to the full history on the demo page. 

**Part of the plugin**

![image](https://github.com/user-attachments/assets/9f63dfdb-82d8-4070-988a-fbbd8691be0c)

----

**Part of the Demo Page**

![image](https://github.com/user-attachments/assets/8e3a47aa-c925-443e-b4d0-07e0baff3235)

**Comment by Jermolene** [comment]
> I think, that the full description of the plugin-history should be part of a demo-edition.
> 
> The plugin itself should only contain the minimal changelog-info. Over time there may be plugins, where the history is much longer than the plugin code itself.

With the scheme I outlined it is entirely up to the plugin publisher how much information they retain.

**Comment by pmario** [comment]
>With the scheme I outlined it is entirely up to the plugin publisher how much information they retain.

That's right. But I was thinking about core plugins. We should also try to keep it minimalistic

**Comment by Jermolene** [comment]
Hi @pmario generally I do not share your concern about wasting space. When and if we encounter the problem where a core plugin has overly length release notes we can consider policies for dealing with it.
